### PR TITLE
PHP 8 Full text search warnings

### PR DESCRIPTION
**Pull request by Eldenroot**
https://community.mybb.com/thread-237723.html

### CODE REVIEW

File: inc/functions_search.php [hunk 1]
```diff
@@ -1709,7 +1709,7 @@ function perform_search_mysql_ft($search)
 
 	// Searching a specific thread?
 	$tidsql = '';
-	if($search['tid'])
+	if(isset($search['tid']) && $search['tid'])
```
Comment (Stefan-MyBB) [inline inc/functions_search.php]: I suggest using `if(!empty($search['tid']))` here.

File: inc/functions_search.php [hunk 2]
```diff
@@ -1727,7 +1727,7 @@ function perform_search_mysql_ft($search)
 	if($search['postthread'] == 1)
 	{
 		// No need to search subjects when looking for results within a specific thread
-		if(!$search['tid'])
+		if(isset($search['tid']) && !$search['tid'])
```
Comment (Stefan-MyBB) [inline inc/functions_search.php]: That looks wrong to me. I suggest using `!empty`.

**Review by Stefan-MyBB — CHANGES_REQUESTED** [review]

**Comment by Eldenroot** [comment]
@Stefan-MyBB - thank you, it should be fixed now

File: inc/functions_search.php [hunk 3]
```diff
@@ -1727,7 +1727,7 @@ function perform_search_mysql_ft($search)
 	if($search['postthread'] == 1)
 	{
 		// No need to search subjects when looking for results within a specific thread
-		if(!$search['tid'])
+		if(!empty($search['tid']))
```
Comment (dvz) [inline inc/functions_search.php]: The logic got modified - `empty()` alone (rather than `!empty()`) would be appropriate in place of a `!` negation

**Review by dvz — CHANGES_REQUESTED** [review]

**Pull request merged**